build error: cannot convert ‘const value_type*’ {aka ‘const char32_t*’} to ‘const char*’

I am trying to build cegui from master and ran into the following issue:
```
[  0%] Building CXX object cegui/src/CMakeFiles/CEGUIBase-9999.dir/MinizipResourceProvider.cpp.o
/home/acxz/vcs/git/github/acxz/pkgbuilds/cegui-git/src/cegui/cegui/src/MinizipResourceProvider.cpp: In member function ‘virtual void CEGUI::MinizipResourceProvider::loadRawDataContainer(const CEGUI::String&, CEGUI::RawDataContainer&, const CEGUI::String&)’:
/home/acxz/vcs/git/github/acxz/pkgbuilds/cegui-git/src/cegui/cegui/src/MinizipResourceProvider.cpp:210:79: error: cannot convert ‘const value_type*’ {aka ‘const char32_t*’} to ‘const char*’
  210 |     if (unzOpenCurrentFilePassword(d_pimpl->d_zfile, d_pimpl->d_password.c_str()) != UNZ_OK)
      |                                                      ~~~~~~~~~~~~~~~~~~~~~~~~~^~
      |                                                                               |
      |                                                                               const value_type* {aka const char32_t*}
In file included from /usr/include/zlib.h:34,
                 from /usr/include/minizip/unzip.h:51,
                 from /home/acxz/vcs/git/github/acxz/pkgbuilds/cegui-git/src/cegui/cegui/src/MinizipResourceProvider.cpp:36:
/usr/include/minizip/unzip.h:337:47: note:   initializing argument 2 of ‘int unzOpenCurrentFilePassword(unzFile, const char*)’
  337 | extern int ZEXPORT unzOpenCurrentFilePassword OF((unzFile file,
      |                                               ^~
make[2]: *** [cegui/src/CMakeFiles/CEGUIBase-9999.dir/build.make:713: cegui/src/CMakeFiles/CEGUIBase-9999.dir/MinizipResourceProvider.cpp.o] Error 1
make[2]: Leaving directory '/home/acxz/vcs/git/github/acxz/pkgbuilds/cegui-git/src/cegui/build'
make[1]: *** [CMakeFiles/Makefile2:1117: cegui/src/CMakeFiles/CEGUIBase-9999.dir/all] Error 2
make[1]: Leaving directory '/home/acxz/vcs/git/github/acxz/pkgbuilds/cegui-git/src/cegui/build'
make: *** [Makefile:152: all] Error 2
```

OS: ArchLinux
GCC: 9.3.0
minizip: 1.2.11
zlib: 1.2.11

--- a/cegui/src/MinizipResourceProvider.cpp
+++ b/cegui/src/MinizipResourceProvider.cpp
@@ -123,7 +123,7 @@ bool MinizipResourceProvider::doesFileExist(const String& filename)
 #if (CEGUI_STRING_CLASS == CEGUI_STRING_CLASS_UTF_8) || (CEGUI_STRING_CLASS == CEGUI_STRING_CLASS_ASCII)
     std::ifstream dataFile(filename.c_str(), std::ios::binary | std::ios::ate);
 #elif CEGUI_STRING_CLASS == CEGUI_STRING_CLASS_UTF_32
-    std::ifstream dataFile(String::convertUtf32ToUtf8(filename.getString()).c_str(), 
+    std::ifstream dataFile(String::convertUtf32ToUtf8(filename.getString()).c_str(),
                            std::ios::binary | std::ios::ate);
 #endif
 
@@ -207,7 +207,7 @@ void MinizipResourceProvider::loadRawDataContainer(const String& filename,
             "' error reading file header");
     }
 
-    if (unzOpenCurrentFilePassword(d_pimpl->d_zfile, d_pimpl->d_password.c_str()) != UNZ_OK)
+    if (unzOpenCurrentFilePassword(d_pimpl->d_zfile, (const char*) d_pimpl->d_password.c_str()) != UNZ_OK)
     {
         throw FileIOException("'" + final_filename +
             "' error opening file");
@@ -259,7 +259,7 @@ size_t MinizipResourceProvider::getResourceGroupFileNames(
 
     char current_name[1024];
     unz_file_info file_info;
-    
+
     if (unzGoToFirstFile(d_pimpl->d_zfile) != UNZ_OK)
     {
         Logger::getSingleton().logEvent(

--- a/cegui/src/RendererModules/OpenGL/GL3Renderer.cpp
+++ b/cegui/src/RendererModules/OpenGL/GL3Renderer.cpp
@@ -50,8 +50,8 @@
 #define BUFFER_OFFSET(i) ((char *)NULL + (i))
 
 #ifdef GLEW_VERSION_4_3
-// The function must be a C method with the same calling convention as the GL API functions, here this is done using the APIENTRY function prefix
-static void APIENTRY OpenGlDebugCallback( GLenum source, GLenum type, GLuint id, GLenum severity, GLsizei length, const GLchar* message, const void* userParam )
+// The function must be a C method with the same calling convention as the GL API functions, here this is done using the GLAPIENTRY function prefix
+static void GLAPIENTRY OpenGlDebugCallback( GLenum source, GLenum type, GLuint id, GLenum severity, GLsizei length, const GLchar* message, const void* userParam )
 {
     std::string str_type;
     switch(type)
@@ -238,7 +238,7 @@ void OpenGL3Renderer::init()
         throw RendererException("Only version 2 and up of OpenGL ES is "
                                 "supported by this type of renderer.");
     initialiseRendererIDString();
-   
+
 #ifdef DEBUG
 #   ifdef GLEW_VERSION_4_3
     if(OpenGLInfo::getSingleton().verAtLeast(4, 3))
@@ -340,14 +340,14 @@ void OpenGL3Renderer::restoreChangedStatesToDefaults(bool isAfterRendering)
 
     if (OpenGLInfo::getSingleton().isPolygonModeSupported())
         glPolygonMode(GL_FRONT_AND_BACK, GL_FILL);
- 
+
     d_openGLStateChanger->disable(GL_CULL_FACE);
     d_openGLStateChanger->disable(GL_DEPTH_TEST);
 
     // During the preparation before rendering, these states will be changed anyways
     // so we do not want to change them extra
     if(isAfterRendering)
-    { 
+    {
         d_openGLStateChanger->disable(GL_BLEND);
         d_openGLStateChanger->disable(GL_SCISSOR_TEST);
     }
@@ -619,7 +619,7 @@ void OpenGL3Renderer::initialiseStandardColouredVAO()
 OpenGLTexture* OpenGL3Renderer::createTexture_impl(const String& name)
 {
     return new OpenGL3Texture(*this, name);
-}  
+}
 
 //----------------------------------------------------------------------------//
 

